.github/workflows/nightlies.yml

```diff
@@ -0,0 +1,310 @@
+      #
+      - name: Run build_all.d for FreeBSD in a dedicated VM
+        if: matrix.target == 'freebsd'
+        uses: vmactions/freebsd-vm@v0.1.3
+        with:
+          usesh: true
+          # Need more RAM than the default 1G
+          mem: 4096
+          prepare: pkg install -y bash curl curlpp git gmake pkgconf gnupg rsync llvm90
+          run: |
+            set -eux
+
+            # Import key used to sign binaries
+            curl https://dlang.org/d-keyring.gpg -o d-keyring.gpg
+            gpg d-keyring.gpg
+
+            # Install ldc
+            curl https://dlang.org/install.sh -o install.sh
+            bash install.sh ldc -p .
+
+            # Use absolute paths because activate doesn't work correctly
+            LDC_BIN=$PWD/ldc-*/bin
+
+            # Determine installed LDC version
+            LDC=$($LDC_BIN/ldc2 --version | head -n 1 | cut -d'(' -f2 | cut -d')' -f1)
+
+            # Determine additional linker flags to make -lcurl work
+            EXTRA_FLAGS="-L$(pkg-config --libs-only-L libcurl)"
+
+            # Actually build the release
+            cd create_dmd_release
+            $LDC_BIN/ldmd2 -g -m64 --link-defaultlib-debug -version=NoVagrant -i build_all.d $EXTRA_FLAGS
+            ./build_all --targets=${{ matrix.target }} "v$LDC" master
+
+      #################################################################
+      # Save the target-specific release as a artifact s.t. the next
+      # job can bundle the entire release for all supported targets
+      #
+      - name: Upload generated release as a temporary artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: dmd-nightly
+          path: |
+            ${{ github.workspace }}/create_dmd_release/build/*
+            !${{ github.workspace }}/create_dmd_release/build/*.zip
+          retention-days: 1
+          if-no-files-found: error
+
+    runs-on: ${{ matrix.os }}
+
+  # Bundles and publishes the entire release
+  generate_release:
+    name: "Publish artifacts on the release page"
+    needs: build-all-releases
+    steps:
+      #################################################################
+      # Fetch all artifacts from the jobs defined above
+      #
+      - name: Download generated releases from the artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: dmd-nightly
+          path: ~/artifacts/
+
+      #################################################################
+      # Debug: Check that all required artifacts are present
+      #
+      - name: Display all files included in the artifacts
+        shell: bash
+        run: |
+          set -euox pipefail
+          ls -aul ~ ~/artifacts
+
+      #################################################################
+      # Create the new release using the downloaded artifacts
+      #
+      - name: Create the nightly release
+        # Currently disabled due to missing permissions
+        if: ${{ false }}
```

Comment:
Doesn't work due to missing permissions because the PR was issued from a fork. Could probably be enabled via a PR from an upstream branch.

See the generated artifacts for a preview of the actual releases